### PR TITLE
Only consider tasks of current version in StartingBehavior

### LINKED ISSUE #829: 0.7.5 - New Deployments Get Stuck
Hello,

We have a large number of apps in our cluster. Sometimes when a new app is pushed, the deployment will get 'stuck' at the RestartApplication phase.

This is not a capacity / resource / constraint problem as our Mesos Masters report adequate memory (30GB+) and adequate CPU (30+). 

When I grep the log for the deployment ID, I initially get the following
```
16475:2014-11-25_18:55:28.42014 [INFO] [11/25/2014 18:55:28.419] [marathon-akka.actor.default-dispatcher-15] [akka://marathon/user/MarathonScheduler/$a/UpgradeManager/88888888-9999-ZZZZ-YYYY-XXXXXXXXXXXX/$b] Killing 0 instances
16476:2014-11-25_18:55:28.42016 [INFO] [11/25/2014 18:55:28.419] [marathon-akka.actor.default-dispatcher-15] [akka://marathon/user/MarathonScheduler/$a/UpgradeManager/88888888-9999-ZZZZ-YYYY-XXXXXXXXXXXX/$b] Successfully killed all the tasks
```
There are 2 containers (tasks) running still, but the log indicates that it's attempting to kill 0 instances.
About 20 to 30 minutes later, the application managed to deploy, a similar message in the logs. Although in this case, the deployment went through only because another leader was elected.

```
2014-11-25_19:16:30.71511 [INFO] [11/25/2014 19:16:30.714] [marathon-akka.actor.default-dispatcher-2] [akka://marathon/user/MarathonScheduler/$a/UpgradeManager/88888888-9999-ZZZZ-YYYY-XXXXXXXXXXXX/$b] Killing 0 instances
2014-11-25_19:16:30.71859 [INFO] [11/25/2014 19:16:30.718] [marathon-akka.actor.default-dispatcher-2] [akka://marathon/user/MarathonScheduler/$a/UpgradeManager88888888-9999-ZZZZ-YYYY-XXXXXXXXXXXX/$b] Successfully killed all the tasks
```

Here is the rest of the log snippet, outside of the akka / dispatcher entries 
https://gist.github.com/sethdmoore/572f017a722fc7d1b964

### PR DESCRIPTION
**Pull request by sttts**
Might fix #829. Please discuss in contrast to #913.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/upgrade/StartingBehavior.scala [hunk 1]
```diff
@@ -69,22 +73,24 @@ trait StartingBehavior { this: Actor with ActorLogging =>
     case MesosStatusUpdateEvent(_, taskId, "TASK_ERROR" | "TASK_FAILED" | "TASK_LOST" | "TASK_KILLED", _, app.`id`, _, _, Version, _, _) => // scalastyle:off line.size.limit
       log.warning(s"Failed to start $taskId for app ${app.id}. Rescheduling.")
       runningTasks -= taskId
+      healthyTasks -= taskId
       taskQueue.add(app)
 
     case Sync =>
-      val actualSize = taskQueue.count(app) + taskTracker.count(app.id)
+      // only count the tasks of this version
+      val actualSize = taskQueue.count(app) + taskTracker.get(app.id).filter(_.getVersion == Version).size
```
Comment (ConnorDoyle) [inline src/main/scala/mesosphere/marathon/upgrade/StartingBehavior.scala]:
```scala
taskTracker.get(app.id).count(...)
```

**Comment by sttts** [comment]
Not sure this PR is correct either. What should happen if a task dies which is not in runningTasks in the StartingBehavior? Shouldn't we add a new task to the taskQueue and increase expectedSizeOfVersion?

**Comment by ConnorDoyle** [comment]
:+1:, I do prefer this approach to #913 because it jives well with the Marathon way (tm), e.g. take actions toward an explicit desired state.  Are there any failing tests to add for the previous behavior?

**Comment by sttts** [comment]
I like that #913 is about the absolute task number. But #913 is not enough as a fix. With #913 the deployment will stop early if tasks die with older version numbers.

**Comment by sttts** [comment]
We have to increate nrToStart when an old task dies. Then I think #913 makes sense.

**Comment by sttts** [comment]
I will prepare an additional patch on top of #913.

**Pull request closed without merging**